### PR TITLE
feat: dropdown-button-icon/custom components

### PR DESCRIPTION
**Pull request by dlockhart**
This introduces two new components which will replace `<d2l-navigation-button-notification-icon>` once [it's removed from sequences](https://github.com/BrightspaceHypermediaComponents/sequences/pull/444) and once the LMS has switches over to these.

dropdown

`<d2l-navigation-dropdown-button-icon>` and `<d2l-navigation-dropdown-button-custom>` differ in that they act as a dropdown opener, which `<d2l-navigation-button-notification-icon>` should have and the LMS needed to do extra work to make happen. They also display tooltips on hover/focus, which has been OK'd by PdM and design and will ship as part of the September release.

### CODE REVIEW

File: d2l-navigation-button-icon.js [hunk 1]
```diff
@@ -121,7 +87,7 @@ class NavigationButtonIcon extends FocusMixin(LitElement) {
 				ariaLabel: this.text,
 				id: this._buttonId,
 				text: nothing,
-				tooltip: html`<d2l-tooltip for="${this._buttonId}" for-type="label" close-on-click>${this.text}</d2l-tooltip>`
+				tooltip: html`<d2l-tooltip close-on-click for="${this._buttonId}" for-type="label" position="bottom">${this.text}</d2l-tooltip>`
```
Comment (dlockhart) [inline d2l-navigation-button-icon.js:90]: We don't ever want these tooltips opening in any direction except down.

File: d2l-navigation-dropdown-button-custom.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { css, html, LitElement } from 'lit';
+import { DropdownOpenerMixin } from '@brightspace-ui/core/components/dropdown/dropdown-opener-mixin.js';
+import { highlightBorderStyles, highlightButtonStyles } from './d2l-navigation-styles.js';
+
+class NavigationDropdownButtonCustom extends DropdownOpenerMixin(LitElement) {
```
Comment (dlockhart) [inline d2l-navigation-dropdown-button-custom.js:5]: This "custom" one is going to be used for the personal menu in the LMS navbar. It has an extra slot for the menu opener contents.

File: d2l-navigation-notification-icon.js [hunk 3]
```diff
@@ -1,44 +1,54 @@
 import '@brightspace-ui/core/components/colors/colors.js';
-import { html, PolymerElement } from '@polymer/polymer/polymer-element.js';
+import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
+import { css, html, LitElement } from 'lit';
```
Comment (dlockhart) [inline d2l-navigation-notification-icon.js:3]: Migrated this to Lit. 🔥

File: d2l-navigation-notification-icon.js [hunk 4]
```diff
@@ -1,44 +1,54 @@
...
-                :host([thin-border]) .d2l-navigation-notification-icon-indicator {
-                    border-width: 1px;
-                }
-            </style>
-            <div class="d2l-navigation-notification-icon-indicator"></div>
+class NavigationNotificationIcon extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			thinBorder: { attribute: 'thin-border', reflect: true, type: Boolean }
```
Comment (dlockhart) [inline d2l-navigation-notification-icon.js:9]: This was missing in the Polymer definition, but is used by the organization tabs stuff.

File: d2l-navigation-styles.js [hunk 5]
```diff
@@ -14,11 +14,50 @@ export const highlightBorderStyles = css`
 		width: calc(100% + 14px);
 	}
 	*:focus > .d2l-navigation-highlight-border,
-	*:hover > .d2l-navigation-highlight-border {
+	*:hover > .d2l-navigation-highlight-border,
+	*[active] > .d2l-navigation-highlight-border {
```
Comment (dlockhart) [inline d2l-navigation-styles.js:18]: When a dropdown opens, it puts `active` on the opener. This ensures that the highlight is displayed when the dropdown is open.

**Comment by svanherk** [comment]
Side note - why are the initials missing in the profile image? 🤔

**Comment by svanherk** [comment]
~~Side note - why are the initials missing in the profile image? 🤔~~

Ha nvm! I thought you were showing the LMS navbar, not the new demo page 🤦

**Review by svanherk — APPROVED** [review]
Looks great!

**Pull request merged**